docker/docker-entrypoint.sh

```diff
@@ -1,20 +1,100 @@
-#!/bin/bash
+#!/bin/sh
+
+function check_migrate() {
+  : "${PROMGEN_DISABLE_MIGRATE:=0}"
+  # These env vars are used as parameters when registering a new shard.
+  # Set defaults:
+  : "${PROMGEN_REGISTER_SHARD:=docker-demo}"
+  : "${PROMGEN_REGISTER_HOST:=prometheus}"
+  : "${PROMGEN_REGISTER_PORT:=9090}"
+
+  if [ ! "${PROMGEN_DISABLE_MIGRATE}" = "1" ]; then
+    set -x
+
+    promgen migrate
+    promgen register "${PROMGEN_REGISTER_SHARD}" "${PROMGEN_REGISTER_HOST}" "${PROMGEN_REGISTER_PORT}"
+
+    set +x
+  fi
+}
+
+function copy_static() {
+  if [ ! -f /srv/promgen/www/static/.present ]; then
+    echo "Copying static files to /srv/promgen/www/static..."
+    set -x
+
+    mkdir -p /srv/promgen/www/static
+    cp -ra /usr/src/app/static /srv/promgen/www
+    touch /srv/promgen/www/static/.present
+
+    set +x
+  fi
+}
 
 set -e
 
 if [ "${1:0:1}" = '-' ]; then
-	set -- promgen "$@"
+  set -- promgen "$@"
 fi
 
 case "$1" in
+web)
+  # Shortcut for launching a Promgen web worker under gunicorn
+  shift
+
+  until promgen check 2>/dev/null
+  do
+    echo "Waiting for database to startup"
+    sleep 3
+  done
+
+  copy_static
+  migrate
+
+  set -- gunicorn "promgen.wsgi:application" --bind 0.0.0.0:8000 "$@"
+  ;;
+web-dev)
+  # Shortcut for launching a Promgen web worker under gunicorn
+  shift
+
+  until promgen check 2>/dev/null
+  do
+    echo "Waiting for database to startup"
+    sleep 3
+  done
+
+  copy_static
+  migrate
+
+  set -- promgen runserver 0.0.0.0:8000 "$@"
+  ;;
 worker)
   # Shortcut to start a celery worker for Promgen
   set -- celery "-A" promgen "$@"
   ;;
-web)
-  # Shortcut for launching a Promgen web worker under gunicorn
+init-config)
   shift
-  set -- gunicorn "promgen.wsgi:application" "$@"
+
+  exec promgen bootstrap --noinput "$@"
```

Comment:
I could be mistaken, but I *think* using `exec` here means that this shell will be replaced by the running program so the later steps will not run.

Reply:
I meant to check this, you're absolutely right, exec just wraps the corresponding syscall.